container-registry-configuration/README.md

```diff
@@ -0,0 +1,189 @@
+
+#### 2.b Using authenticated accounts
+
+We can pass credentials to `containerd` configuration to be not limited by registry provider limits.
+
+#### Decision
+
+`2.b` is selected.
+We will use authenticated accounts.
+
+
+### 3. Registries
+
+The infranet page (See `Registry Mirrors`) states this as a desired configuration:
+  - `docker.io` as primary public registry since it has a privilege in docker daemon.
+  - Our own Azure Container Registry as secondary registry because of the security concerns
+
+Since we use `containerd`, docker has not a privelege anymore but there is no need to change this order at the moment. We will follow this configuration for CAPI clusters too.
+
+### 4. Accounts
+
+- 4.a Giantswarm Account + Single account
+- 4.b Giantswarm Account + Per MC including its WCs
+- 4.c Giantswarm Account + Per WC
+- 4.b Customer Accounts
+
+#### Decision
+
+`4.b` is selected.
+
+Since it is a part of the platform itself, we are going to use GiantSwarm accounts in containerd configuration.
+As we do for other operators, we will follow "per MC" approach here to.
+
+### 5. How to provide credentials to WCs
+
+#### 5.a cluster-apps-operator
+
+We can propogate credentials from MC to WC by using `cluster-apps-operator`. It can create a secret per WC with the credentials.
+
+#### 5.b Optional interface + Gitops
+
+We can delegate the responsibility of creating credentials secret to the creator of `cluster-$provider` apps. GitOps can be useful to template the secrets per WC.
+
+#### Decision
+
+`5.b` is selected.
+
+We don't want to add another responsility to `cluster-apps-operator`. 
+
+
+### 6. Default configuration for WCs
+
+- 6.a: Configuring WCs by default
+- 6.b: Not configuring WCs by default
+
+#### Decision
+
+**NOT DECIDED YET**
+
+
+### 7. Where to put the credentials
+
+#### 7.a Customer's Git Repository
+
+We can put the secret into customers' git repositories. Customers will access to our credentials.
+
+#### 7.b Management Cluster Fleet
+
+We can put the secret into our `management-clusters-fleet` repo. We will need to give reference to that secret in customers' git repository. Management of these secrets can be so costly when we need to create the secret in each organization namespace.
+
+#### Decision
+
+**NOT DECIDED YET**
+
+
+### 8. Configuration interface
+
+How will be the configuration interface in `cluster-$provider` apps?
+
+#### 8.a Full containerd configuration
+
+We can implement a full transitive configuration interface so that users can provide a full containerd configuration, including registy credentials.
+
+#### 8.b Only registry credentials in a structred way
+
+We can define a configuration interface like below and render containerd configuration in `cluster-$provider` app chart.
+
+```
+registries:
+- name: docker
+  endpoint: https://registry-1.docker.io
+  username:
+  password:
+- name: azurecr
+  endpoint: giantswarm.azurecr.io
+  ...
+```
```

Comment:
Even (you decide)[https://github.com/giantswarm/rfc/pull/49/files#r1027860367] use the same credentials for all accounts I would still allow customers to add their own mirrors. It is a requirement by Panamax that we implemented in KVM via k8s initiator app

Reply:
I was just today in a meeting with Panamax where they said that it is a requirement in the future that all images are pulled through their own registry mirror (artifactory).

Reply:
Requirement in the future is totally fine. That's a function for Harbour.